File: server/src/models/Chapter.js
const mongoose = require('mongoose');

const chapterSchema = new mongoose.Schema({
  number: {
    type: Number,
    required: true,
    unique: true,
    min: 1,
    max: 18,
  },
  titleSanskrit: {
    type: String,
    required: true,
  },
  titleEnglish: {
    type: String,
    required: true,
  },
  titleHindi: {
    type: String,
    required: true,
  },
  description: {
    type: String,
    required: true,
  },
  verseCount: {
    type: Number,
    required: true,
  },
  imageUrl: {
    type: String,
    default: '',
  },
  videoUrl: {
    type: String,
    default: '',
  },
  keyThemes: [{
    type: String,
  }],
  totalActivities: {
    type: Number,
    default: 7,
  },
}, {
  timestamps: true,
  toJSON: { 
    virtuals: true,
    transform: function(doc, ret) {
      ret.id = ret.number.toString();
      return ret;
    }
  },
  toObject: { virtuals: true },
});

module.exports = mongoose.model('Chapter', chapterSchema);

File: backup/geeta book app copy 2/lib/shared/domain/models/chapter.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'chapter.freezed.dart';
part 'chapter.g.dart';

@freezed
class Chapter with _$Chapter {
  const factory Chapter({
    required String id,
    required int number,
    required String titleSanskrit,
    required String titleEnglish,
    required String titleHindi,
    required String description,
    required int verseCount,
    required String imageUrl,
    String? videoUrl,
    @Default([]) List<String> keyThemes,
    @Default(false) bool isCompleted,
    @Default(0) int completedActivities,
    @Default(0) int totalActivities,
  }) = _Chapter;

  factory Chapter.fromJson(Map<String, dynamic> json) =>
      _$ChapterFromJson(json);
}
